awscli/customizations/wizard/ui/prompt.py

```diff
@@ -124,6 +136,60 @@ def __pt_container__(self):
         return self.container
 
 
+class WizardPromptCompletionAnswer(WizardPromptAnswer):
+    COMPLETION_MENU_MAX_HEIGHT = 5
+
+    def __init__(self, value_name, default_value=None, completer=None):
+        self._completer = self._get_completer(completer)
+        super().__init__(value_name, default_value)
+
+    def _get_completer(self, completer):
+        return {
+            'file_completer': PathCompleter(expanduser=True)
+        }[completer]
+
+    def _get_answer_buffer(self):
+        return Buffer(name=self._value_name,
+                      completer=self._completer,
+                      complete_while_typing=True,
+                      document=Document(text=self._default_value))
+
+    def _get_menu_height(self):
+        if self._buffer.complete_state:
+            return min(len(self._buffer.complete_state.completions),
+                       self.COMPLETION_MENU_MAX_HEIGHT)
+        return 0
+
+    def _get_answer_container(self):
+        return FloatContainer(
+            FullyExtendedWidthWindow(
+                content=BufferControl(buffer=self._buffer),
+                style=self._get_style,
+                wrap_lines=True
+            ),
+            floats=[
+                Float(
+                    xcursor=True, ycursor=True, top=1,
+                    height=self._get_menu_height,
+                    content=CompletionsMenu(),
+                )
+            ],
+            key_bindings=self._get_key_bindings()
```

Comment:
One thing that tripped me up using it was that if you hit `<TAB>` while completing it will not toggle to the suggested completion and instead move onto the next prompt. I'm wondering if we should override the tab keybindings for anything that has a completer? I'm open to ideas on this.